CIPs/0003.md

```diff
@@ -0,0 +1,134 @@
+
+There are a number of principals that provide resources to or deploy services on the Celo platform that may wish to assert identities or provide metadata. These include, but are not limited to:
+* Validators
+* Validator groups
+* Smart contract owners
+* DApp developers
+* Creators of Governance proposals
+* Oracles 
+
+Use cases include:
+* Tools want to present public metadata supplied by a validator or validator group as part of a list of candidate groups, or a list of current elected validators.
+* Governance Explorer UIs may want to present public metadata about the creators of governance proposals
+* The Celo Foundation receives notice of a security vulnerability and wants to contact elected validators to facilitate them to make a decision on applying a patch.
+* A DApp makes a request to the Celo Wallet for account information or to sign a transaction. The Celo Wallet should provide information about the DApp to allow the user to make a decision whether to accept the transaction or not.   
+
+## Goals
+
+- Integrate with the existing Attestations structure, rather than maintain separate identifer fields in other contracts, so that the same identity claims and metadata can be linked from any of a principals' activities  
+- Allow principals to make an extensible range of identity claims, and support clients verifying the veracity of these 
+- Allow principals to supply unverified extensible, structured metadata off-chain
+- Allow tools to choose what metadata they require and how they present it
+- Support both public (unencrypted) metadata and metadata accessible only to the Celo Foundation
+
+## Proposed Solution
+
+### Claims
+
+The Attestations contract is extended as follows:
+*  Add a `struct Claim` that comprises: 
+   * `service` - Required. Initially supported values for `service` include `primary`, `github`, and `twitter`.
+   * `identifier` - Optional. The identifier being claimed on the service being claimed.  
+   * `data` - Optional. Additional data, e.g. a 
+   URL, to locate a linked signed reference to the claim. 
+* A field array `Claim[] claims` is added to the `Account` struct. 
+
+All claims are optional.
+
+### Primary Claims and Public Metadata
+
+For claims with `service` set to `primary`, `identifier` is a domain name. `data` is an optional path prefix, such that `https://<identifier>/<data>/identity.json` is a file in the following format:
+
+```
+{
+  "address": "<address>",
+  "claim": {
+    "service": "primary",
+    "identifier": "..",
+    "ctime": 1231222312,
+    "expiry" : 31536000, 
+  },
+  "signature": "<sig>",
+}
+```
+
+TODO - detail exact format. 
+
+Clients should verify the SSL/TLS certificate against a chain of trust and check it was issued for the domain and `identifier` value in question. 
+
+Public metadata for a principal with a `primary` claim may be obtained by first verifying the claim then retrieving the resource at `<url>/<address>-metadata.json`.
```

Comment:
Do we need the metadata to be hosted
1. Under the actual domain
2. Under a specified path

Mostly as a personal anecdote, but with my personal setup at nambrot.com, it would kind of a pain to host it on my specific path. I'd rather prefer to be able to specify under which URL (any domain) I would like to host my public metadata at. What do you think about just adding a `metadata` field to the `Account` struct

It separates the concern of proving a claim from hosting data with the identity of that claim.